spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/spanner/GcpSpannerAutoConfigurationTests.java

```diff
@@ -107,6 +110,24 @@ public void testIdConverterNotCreated() {
 				.run(context -> assertThat(context.getBeansOfType(BackendIdConverter.class)).isEmpty());
 	}
 
+	@Test
+	public void testSpannerCustomizerProvided() {
+		Duration duration = Duration.ofSeconds(42);
+		this.contextRunner.withBean(SpannerOptionsCustomizer.class, () -> {
+			return builder -> {
+				builder.getSpannerStubSettingsBuilder()
+						.executeSqlSettings()
+						.setRetrySettings(RetrySettings.newBuilder().setMaxRetryDelay(duration).build());
+			};
+		}).run(context -> {
+			SpannerOptions spannerOptions = context.getBean(SpannerOptions.class);
```

Comment:
Ignore this if it seems like overkill, but do you think it would be worth adding some assertion that the "core" configuration of `SpannerOptions` provided by `GcpSpannerAutoConfiguration` has been preserved? For example (though maybe getting too far into implementation details), if there were a way to set or mock `CoreSpannerAutoConfiguration.numRpcChannels`, and then assert that `spannerOptions.getNumChannels()` has the same value?

Reply:
I am always in favor of more testing.